File: projectEuler/20/20.py
#Factorial Digit Sum

def factorial(n):
    if n ==0:
        return 1
    else:
        return n*factorial(n-1)
    
def sum_of_digits(n):
    sum = 0
    #convert n to string to iterate over each digit
    for i in str(n):
        #convert each digit back to int and add to sum
        sum += int(i)
    return sum

print(sum_of_digits(factorial(100)))

File: projectEuler/20/20.cpp
#include <iostream>
#include <vector>
#include <numeric>


using namespace std;

// Function to multiply a large number (represented as a vector) by an integer
void multiply(vector<int>& number, int n) {
    int carry = 0;
    for (auto& digit : number) {
        int product = digit * n + carry;
        digit = product % 10;
        carry = product / 10;
    }
    while (carry) {
        number.push_back(carry % 10);
        carry /= 10;
    }
}

// Function to calculate the sum of digits of factorial
int factorialDigitSum(int n) {
    vector<int> fact = {1}; // Initialize with factorial of 0
    for (int i = 1; i <= n; ++i) {
        multiply(fact, i);
    }
    return accumulate(fact.begin(), fact.end(), 0);
}

int main() {
    cout << factorialDigitSum(100) << endl;
    return 0;
}